src/lib/queries/finding/addFinding.ts

```diff
@@ -1,18 +1,50 @@
 import {MutateOptions, useMutation} from '@tanstack/react-query'
+import {upload} from '@vercel/blob/client'
 
 import {
-  AddFindingRequest,
+  AddFinding,
   AddFindingResponse,
   addFinding,
 } from '@/server/actions/finding/addFinding'
 import {withApiErrorHandler} from '@/lib/utils/common/error'
 
+type AddFindingRequest = {
+  finding: AddFinding
+  attachments?: File[]
+}
+
+const addFindingWithAttachments = async ({
```

Comment:
Wouldn't it be better to move this to the actions folder rather than the queries?

Reply:
It's using [client side upload](https://vercel.com/docs/storage/vercel-blob/client-upload), so it runs in browser (it uploads the files to vercel blob storage directly from browser), so I think it better fits here, the `actions` folder should be just for server actions (as if it would be an API).

Reply:
Alright, let's keep it there then